domain/operation/service/task.go

```diff
@@ -0,0 +1,27 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package service
+
+import (
+	"context"
+
+	"github.com/juju/juju/core/errors"
+	"github.com/juju/juju/domain/operation"
+)
+
+// StartTask marks an action as running, and logs the time it was started.
+func (s *Service) StartTask(ctx context.Context, id string) error {
```

Comment:
Maybe this was already discussed but I'm not aware of the discussion result. 
Why not adding this to `action.go` as `BeginAction()` or `StartAction()`? 
I might be wrong, but in my mind we were trying not to leak the "task" concept outside the domain, and the contract would be start/finish of operations (actions/execs).
Also, this method is receiving an action tag from the facade, which indicates it's actually an action.

Am I missing something?

Reply:
There is no reason to keep `operation` and `task` concepts limited to the domain. In fact, the BeginAction in the `machinactions` worker is in fact BeginTask as machines do not have actions. It. was written that way due to the common apiserver code previously written.

An operation is an umbrella for the individual units/machines running something as a group. Neither the unit nor machine cares about the overall operation, they care about what they need to do specifically which is the task.

Reply:
Ok I understand, but then what will this method do? 
Just fire one single task from an operation? Or start the action (like the function doc says), i.e. the first task in the action (operation)?